runtime/src/dispatch.rs

```diff
@@ -0,0 +1,180 @@
+use castaway::cast;
+use std::marker::PhantomData;
+
+use fvm_ipld_encoding::ipld_block::IpldBlock;
+use serde::{Deserialize, Serialize};
+
+use crate::ActorError;
+
+/// Implement actor method dispatch:
+///
+/// ```ignore
+/// type Actor;
+/// #[derive(FromPrimitive)]
+/// #[repr(u64)]
+/// enum Method {
+///     Constructor = 1,
+/// }
+/// impl ActorCode for Actor {
+///     type Methods = Method;
+///     actor_dispatch! {
+///         Constructor => constructor,
+///     }
+/// }
+/// ```
+#[macro_export]
+macro_rules! actor_dispatch {
+    ($($method:ident => $func:ident,)*) => {
+        fn invoke_method<RT>(
+            rt: &mut RT,
+            method: MethodNum,
+            args: Option<fvm_ipld_encoding::ipld_block::IpldBlock>,
+        ) -> Result<Option<fvm_ipld_encoding::ipld_block::IpldBlock>, ActorError>
+        where
+            RT: Runtime,
+            RT::Blockstore: Clone,
+        {
+            restrict_internal_api(rt, method)?;
+            match FromPrimitive::from_u64(method) {
+                $(Some(Self::Methods::$method) => $crate::dispatch(rt, Self::$func, &args),)*
+                None => Err(actor_error!(unhandled_message; "invalid method: {}", method)),
```

Comment:
`unhandled_message` is not imported in this macro? That means the caller needs import it? Or we can just specify the full path.